subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -1134,14 +1138,28 @@ void ull_scan_aux_release(memq_link_t *link, struct node_rx_pdu *rx)
 		lll = rx->rx_ftr.param;
 		lll_aux = rx->rx_ftr.lll_aux;
 
+		/* Under race condition when LLL scheduling a reception of
+		 * auxiliary PDU, a scan aux context may be assigned late and
+		 * the node rx releasing the aux context will not have it.
+		 * Release the scan aux context assigned in the scan context.
+		 */
+		if (!lll_aux) {
+			lll_aux = lll->lll_aux;
+		}
```

Comment:
I'm not following the comment. It states "Release the scan aux context assigned in the scan context." but there's only an assignment here?